rules/0131.go

```diff
@@ -0,0 +1,202 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rules
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/descriptor"
+	p "google.golang.org/protobuf/reflect/protoreflect"
+)
+
+func init() {
+	registerRules(
+		checkGetRequestMessageName(),
+		checkGetRequestMessageNameField(),
+		checkGetRequestMessageUnknownFields(),
+		checkGetResponseMessageName(),
+	)
+}
+
+// Get messages should have a properly named Request message.
+func checkGetRequestMessageName() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0131", "request_message", "name"),
+			Description:  "check that Get RPCs have appropriate request messages",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#guidance",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get methods for the purpose of this rule;
+				// ignore everything else.
+				if !isGetMethod(m) {
+					return
+				}
+
+				// Rule check: Establish that for methods such as `GetFoo`, the request
+				// message is named `GetFooRequest`.
+				methodName := string(m.Name())
+				requestMessageName := string(m.Input().Name())
+				if requestMessageName != methodName+"Request" {
+					problems = append(problems, lint.Problem{
+						Message: fmt.Sprintf(
+							"Get RPCs should have a request message named after the RPC, such as %q.",
+							methodName+"Request",
+						),
+						Suggestion: methodName + "Request",
+						Descriptor: m,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// The Get standard method should only have expected fields.
+func checkGetRequestMessageNameField() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0131", "request_message", "name_field"),
+			Description:  "check that a name field is present",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/131#request-message",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get methods for the purpose of this rule;
+				// ignore everything else.
+				if !isGetMethod(m) {
+					return
+				}
+
+				// Rule check: Establish that a name field is present.
+				nameField := m.Input().Fields().ByName("name")
+				if nameField == nil {
+					problems = append(problems, lint.Problem{
+						Message:    fmt.Sprintf("method %q has no `name` field", m.Name()),
+						Descriptor: m.Input(),
```

Comment:
The lint warning should show up in the RPC method, not in the input message, i.e., we should return `m` here.

Reply:
`m.Input()` is correct. The problem is on the input message, not the method; having the lint warning show up on the RPC would be more difficult for the producer.

Reply:
If a user wants to disable this rule, it will be on the input message as well, right? Currently, we are using the returned descriptor (here is the `m.Input()`) to check if the rule is disabled on the leading comments of the descriptor.